### PR TITLE
[MPS] Add 2d grid sampler

### PR DESCRIPTION
**Pull request by DenisVieriu97**
Add support for MPS grid sampler

### CODE REVIEW

File: aten/src/ATen/mps/MPSDevice.mm [hunk 1]
```diff
@@ -90,8 +90,17 @@ static inline MTLLanguageVersion getMetalLanguageVersion(const id<MTLDevice>& de
...
-bool MPSDevice::isMacOS13Plus() const {
-  return _macos13plus;
+bool MPSDevice::isMacOS13Plus(int32_t subVersion) const {
+  id mpsCD = NSClassFromString(@"MPSGraph");
+  static bool _macos_13_0_plus = [mpsCD instancesRespondToSelector:@selector(cumulativeSumWithTensor:axis:name:)] == YES;
+  static bool _macos_13_1_plus = [mpsCD instancesRespondToSelector:@selector(
+    sampleGridWithSourceTensor:coordinateTensor:layout:normalizeCoordinates:relativeCoordinates:alignCorners:paddingMode:samplingMode:constantValue:name:)] == YES;
+
+  switch (subVersion) {
+    case 0:  return _macos_13_0_plus;
```
Comment (razarmehr) [inline aten/src/ATen/mps/MPSDevice.mm:100]: Can we replace these raw numbers 0, 1, etc., with enums like `MACOS_VER_13_0_PLUS`, `MACOS_VER_13_1_PLUS`, and etc.?

Comment (razarmehr) [reply aten/src/ATen/mps/MPSDevice.mm:100]: Or you'll do that in a separate PR?

Comment (kulinseth) [reply aten/src/ATen/mps/MPSDevice.mm:100]: Lets add in this PR.

File: test/test_mps.py [hunk 2]
```diff
@@ -7187,6 +7187,244 @@ def test_conv2d_single_stride(self):
             x_gpu = conv_gpu(y_gpu)
             self.assertEqual(x_cpu, x_gpu.cpu(), rtol=1e-03, atol=1e-05)
 
+    def test_grid_sample(self):
```
Comment (kulinseth) [inline test/test_mps.py:7190]: There is no test in TestConsistency ?

**Review by kulinseth — CHANGES_REQUESTED** [review]

File: test/test_mps.py (hunk 2 above)
Comment (DenisVieriu97) [reply test/test_mps.py:7190]: TestConsistency uses border padding which is unsupported through MPS at this moment.

File: aten/src/ATen/mps/MPSDevice.mm (hunk 1 above)
Comment (DenisVieriu97) [reply aten/src/ATen/mps/MPSDevice.mm]: Thanks. Updated

**Review by razarmehr — APPROVED** [review]
LGTM

File: aten/src/ATen/native/mps/operations/GridSampler.mm [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+    runMPSGraph(stream, cachedGraph->graph(), feeds, results);
+  }
+#endif // defined(__MAC_13_2)
+}
+
+Tensor grid_sampler_2d_mps(const Tensor& input, const Tensor& grid,
+                           int64_t interpolation_mode, int64_t padding_mode,
+                           bool align_corners) {
+#if defined(__MAC_13_2)
```
Comment (malfet) [inline aten/src/ATen/native/mps/operations/GridSampler.mm:131]: This means that GridSample is likely not available in any of our binary builds... :/

Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/operations/GridSampler.mm:131]: Interesting. There is a test added for grid sampler which should fail if that is the case. I'll take a look

**Pull request closed without merging**